### DISCUSSION TITLE
API health check

### DISCUSSION THREAD

**Discussion by RickOttenNS** [Azure/api-management-developer-portal#2602] [Feature request]
We have multiple API teams that would like to display the health of their APIs to (potential) consumers. A simple stoplight-function would suffice. Ideally, this feature would be integrated within the developer portal. 

I suppose there are multiple ways to go about this. A health page with all APIs a consumer is subscribed to would be ideal, but the developer portal would have to facilitate this and besides: there is no generic GET /health endpoint per API. Even if there was, a generic /health call would only show whether APIM is up or not (for which the existing /status-0123456789abcdef would suffice) or if an API is deployed or not. Using metrics instead might be an option, but it's less reliable with APIs that see less usage. 

Are there any plans to facilitate in this within the hosted dev portal, or is there documentation available on how to achieve this?

**Comment by ghost** [comment]
@RickOttenNS, thank you for opening this issue. We will triage it within the next few business days.

**Comment by mikebudzynski** [comment]
> Are there any plans to facilitate in this within the hosted dev portal, or is there documentation available on how to achieve this?

We don't have immediate plans to offer this functionality in the developer portal. As a workaround, you can integrate 3rd party API health monitoring solutions with [custom widgets or HTML code injection](https://learn.microsoft.com/azure/api-management/developer-portal-extend-custom-functionality).

**Comment by ghost** [comment]
@RickOttenNS, thank you for requesting this feature.
